crates/astria-core/src/execution/v1alpha2/mod.rs

```diff
@@ -2,9 +2,119 @@ use prost_types::Timestamp;
 
 use crate::{
     generated::execution::v1alpha2 as raw,
+    sequencer::v1alpha1::{
+        IncorrectRollupIdLength,
+        RollupId,
+    },
     Protobuf,
 };
 
+// An error when transforming a [`raw::GenesisInfo`] into a [`GenesisInfo`].
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct GenesisInfoError(GenesisInfoErrorKind);
+
+impl GenesisInfoError {
+    fn incorrect_rollup_id_length(inner: IncorrectRollupIdLength) -> Self {
+        Self(GenesisInfoErrorKind::IncorrectRollupIdLength(inner))
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum GenesisInfoErrorKind {
+    #[error("`rollup_id` field did not contain a valid rollup ID")]
+    IncorrectRollupIdLength(IncorrectRollupIdLength),
+}
+
+/// Genesis Info required from a rollup to start a an execution client.
+///
+/// Contains information about the rollup id, and base heights for both sequencer & celestia.
+///
+/// Usually constructed its [`Protobuf`] implementation from a
+/// [`raw::GenesisInfo`].
+#[derive(Clone, Copy, Debug)]
+pub struct GenesisInfo {
+    /// The rollup id which is used to identify the rollup txs.
+    rollup_id: RollupId,
+    /// The first height of sequencer which is used to identify the first block of the rollup.
+    sequencer_genesis_block_height: tendermint::block::Height,
+    /// The first height of celestia which to look for sequencer blocks.
+    celestia_base_block_height: celestia_tendermint::block::Height,
```

Comment:
is this the only place `celestia_tendermint` is used? is their height type internally a different type than normal tendermint?

Reply:
The type is internally exactly the same, but unfortunately from a type-system perspective it's different.